R/generateFilterValues.R

```diff
@@ -77,21 +85,88 @@ generateFilterValuesData = function(task, method = "randomForestSRC.rfsrc", nsel
     x[match(fn, names(x))]
   })
 
-  fval = do.call(cbind, fval)
-  colnames(fval) = method
-  types = vcapply(getTaskData(task, target.extra = TRUE)$data[fn], getClass1)
-  out = data.frame(name = row.names(fval),
-                   type = types,
-                   fval, row.names = NULL, stringsAsFactors = FALSE)
+  ### ensemble
+
+  if (!is.null(ensemble.method)) {
+
+    # format and order simple filter methods
+    fval_all = set_names(fval, method) %>%
+      map(~ cbind(.x)) %>%
+      map(~ as.data.frame(.x)) %>%
+      map(~ rename(.x, value = .x)) %>%
+      map(~ mutate(.x, name = rownames(.x))) %>%
+      map(~ arrange(.x, desc(.x$value))) %>%
+      map(~ mutate(.x, type = vcapply(getTaskData(task, target.extra = TRUE)$data[fn], getClass1)))
+
+    # rank and merge all information together into one data.frame
+    fval_all_ranked = map(fval_all, ~ mutate(.x, rank = seq(1:length(.x$value)))) %>%
+      imap(~ mutate(.x, method = names(fval_all[.y]))) %>%
+      map_dfr(~ as.data.frame(.x))
+
+    ### ensemble rank aggregation
+
+    if (any(c("E-min", "E-mean", "E-median", "E-max", "E-Borda") %in% ensemble.method)) {
```

Comment:
Possible values for ensemble method should be documented.

Reply:
Yep, I will ofc do this.